### PR TITLE
[IMP] l10n_in: enable cash rounding

### PR DESCRIPTION
**Pull request by zepa-odoo**
Implement default Cash Rounding with Half-Up method for Indian localization. Update Chart of Accounts for Expense & Income with appropriate adjustments.

Task Id: 3631283

### CODE REVIEW

File: addons/l10n_in/__init__.py [hunk 1]
```diff
@@ -2,3 +2,20 @@
...
+    '''
+    # The cash rounding is activated by default only if the company is localized in Switzerland or Liechtenstein.
+    for company in env['res.company'].search([('partner_id.country_id.code', '=', "IN")]):
+        config_wizard = env['res.config.settings'].create({
+            'company_id': company.id,
+            'group_cash_rounding': True
+        })
+        # We need to call execute, otherwise the "implied_group" in fields are not processed.
+        config_wizard.execute()
+        config_wizard.unlink()
```
Comment (jva-odoo) [inline addons/l10n_in/__init__.py:18]:
```suggestion
    group_user = self.env.ref('base.group_user').sudo()
    group_user._apply_group(self.env.ref('account.group_cash_rounding'))
```

Comment (zepa-odoo) [reply addons/l10n_in/__init__.py]: Done

**Review by rugo-odoo — CHANGES_REQUESTED** [review]

**Comment by rugo-odoo** [comment]
(cc @william-andre)

File: addons/l10n_in/models/template_in.py [hunk 2]
```diff
@@ -38,3 +38,10 @@ def _get_in_res_company(self):
                 'account_purchase_tax_id': 'igst_purchase_18',
             },
         }
+
+    def _post_load_data(self, template_code, company, template_data):
+        self.env.ref('l10n_in.cash_rounding_in_half_up').write({
+            'profit_account_id': self.env.ref('account.%s_p213202'%(self.env.company.id), raise_if_not_found=False),
+            'loss_account_id': self.env.ref('account.%s_p213201'%(self.env.company.id), raise_if_not_found=False)
+        })
+        return super()._post_load_data(template_code, company, template_data)
```
Comment (william-andre) [inline addons/l10n_in/models/template_in.py]:
```suggestion
    @template('in', 'account.cash.rounding')
    def _get_in_account_cash_rounding(self):
        return {
            'l10n_in.cash_rounding_in_half_up': {
                'profit_account_id': 'p213202',
                'loss_account_id': 'p213201',
            }
        }
```

File: addons/account/views/json of gstr2a
Comment (william-andre) [inline addons/account/views/json of gstr2a:1]: hum

File: addons/l10n_in/__init__.py [hunk 3]
```diff
@@ -2,3 +2,14 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import models
+
+
+def init_settings(env):
+    """If the company is localized in India, activate the cash rounding by default.
+    """
+    # The cash rounding is activated by default only if the company is localized in India.
```
Comment (william-andre) [inline addons/l10n_in/__init__.py:10]: The comment is confusing. It is not for **only** for Indian companies, but as soon as the module is installed, for **all** the companies.

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**